crates/bevy_render/src/render_graph/nodes/render_resources_node.rs

```diff
@@ -808,3 +808,192 @@ where
 
     success
 }
+
+#[derive(Debug, Default)]
+pub struct GlobalRenderResourcesNode<T>
+where
+    T: RenderResources,
+{
+    command_queue: CommandQueue,
+    marker: PhantomData<T>,
+}
+
+impl<T> GlobalRenderResourcesNode<T>
+where
+    T: RenderResources,
+{
+    pub fn new() -> Self {
+        Self {
+            command_queue: CommandQueue::default(),
+            marker: Default::default(),
+        }
+    }
+}
+
+impl<T> Node for GlobalRenderResourcesNode<T>
+where
+    T: RenderResources,
+{
+    fn update(
+        &mut self,
+        _world: &World,
+        render_context: &mut dyn RenderContext,
+        _input: &ResourceSlots,
+        _output: &mut ResourceSlots,
+    ) {
+        // dbg!();
+        self.command_queue.execute(render_context);
+    }
+}
+
+impl<T> SystemNode for GlobalRenderResourcesNode<T>
+where
+    T: renderer::RenderResources,
+{
+    fn get_system(&self) -> BoxedSystem {
+        let system = global_render_resources_node_system::<T>
+            .system()
+            .config(|config| {
+                config.0 = Some(GlobalRenderResourcesNodeState {
+                    command_queue: self.command_queue.clone(),
+                    ..Default::default()
+                })
+            });
+
+        Box::new(system)
+    }
+}
+
+struct GlobalRenderResourcesNodeState<T: RenderResources> {
+    command_queue: CommandQueue,
+    staging_buffer: Option<BufferId>,
+    target_buffer: Option<BufferId>,
+    _marker: PhantomData<T>,
+}
+
+impl<T: RenderResources> Default for GlobalRenderResourcesNodeState<T> {
+    fn default() -> Self {
+        Self {
+            command_queue: Default::default(),
+            staging_buffer: None,
+            target_buffer: None,
+            _marker: Default::default(),
+        }
+    }
+}
+
+fn global_render_resources_node_system<T: RenderResources>(
+    mut state: Local<GlobalRenderResourcesNodeState<T>>,
+    render_resources: Res<T>,
+    render_resource_context: Res<Box<dyn RenderResourceContext>>,
+    mut render_resource_bindings: ResMut<RenderResourceBindings>,
+) {
+    // TODO add support for textures
+    // No need to do anything if no changes
+    if render_resources.is_changed() {
+        // Precalculate the aligned size of the whole render resources buffer
+        let aligned_size = render_resources
+            .iter()
+            .fold(0, |aligned_size, render_resource| {
+                aligned_size
+                    + render_resource_context
+                        .get_aligned_uniform_size(render_resource.buffer_byte_len().unwrap(), false)
+            });
+
+        // Get old buffer and possibly resize
+        let staging_buffer = if let Some(staging_buffer) = state.staging_buffer {
```

Comment:
This is a bit of a style nit, but using `match` over `if let Some(_) = option { } else { }` is much clearer.

```
let staging_buffer = match state.staging_buffer {
    Some(staging_buffer) => {}
    None => {}
}
```

Same with lines 930-953.

Reply:
Done.